exercises/06-b/README.md

```diff
@@ -0,0 +1,317 @@
+
+- https://docs.particular.net/nservicebus/hosting/cloud-services-host/critical
+
+
+### Step 6
+
+Repeat the previous steps for the *Sales* and *Shipping* projects.
+
+
+## Exercise 6.2: Running installers only during setup/installation
+
+The current endpoints always calls `endpointConfiguration.EnableInstallers()` which is not recommended for production. Processes often need additional permissions to create queues and storage schemas and in production it is prefered to run least privileged.
+
+### Step 1
+
+Open `Program.cs` of Billing.
+
+### Step 2
+
+Remove the line `endpointConfiguration.EnableInstallers()`.
+
+### Step 3
+
+Copy the following code to the endpoint configuration code.
+
+```c#
+if (Environment.UserInteractive && Debugger.IsAttached)
+{
+    endpointConfiguration.EnableInstallers();
+}
+```
+
+### Step 4
+
+Repeat step 2 and 3 for *Sales*, *Shipping* and *ClientUI*.
+
+
+## Exercise 6.2: Configure a custom logging framework
+
+The default logging feature only logs to a file and does not support filtering. In this exercise we are going to configure our endpoints to use **NLog** through the Microsoft.Extensions, and write log entries to a file on a path different then the deployed executable.
+
+### Step 1
+
+Add nuget package `NLog.Extensions.Hosting` at the solution level to all projects.
+
+### Step 2
+
+Open the `Program.cs` from *Billing*.
+
+### Step 2
+
+Add the following code to the correct location in the currently opened `Program.cs`:
+
+```c#
+builder.UseNLog();
+```
+
+### Step 3
+
+Run the Billing endpoint and notice startup fails. This is because NLog cannot find its configuration.
+
+### Step 4
+
+Create an `nlog.config` and add the following NLog related snippets to the configuration.
+
+```xml
+<?xml version="1.0" encoding="utf-8" ?>
+<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
+      autoReload="true"
+      throwExceptions="false"
+      internalLogLevel="Off" internalLogFile="c:\temp\nlog-internal.log">
+
+
+    <targets>
+        <target xsi:type="File"
+                name="mainLog"
+                fileName="C:\logs\nlog-billing-${shortdate}.log"
+                layout="${longdate} level=${level} source=${callsite} message=${message}"
+                keepFileOpen ="false"
+                concurrentWrites ="true"
+                archiveNumbering="DateAndSequence"
+                archiveAboveSize="1000000"
+                maxArchiveFiles="10"/>
+    </targets>
+
+    <rules>
+        <logger name="*" minlevel="Trace" writeTo="mainLog" />
+    </rules>
+
+</nlog>
+```
+
+### Step 5
+
+Run the Billing endpoint. You should now see Debug output in the console window. Also check the folder on `C:/logs` and see if it contains a file called `nlog-shipping-date.log`.
+
+### Step 6
+
+Repeat the same steps for the rest of the projects.
```

Comment:
Is there learning value in repeating the steps? This will take students quite some time already and there's not that much time.

Reply:
I don't think this matters. They get a certain amount of time for the exercises, if they can get it done in that time frame, then why not? There will be others in the group that might not reach this step at all due to lack of time.